compass/landice/tests/humboldt/mesh.py

```diff
@@ -0,0 +1,347 @@
+            ((3, 0), 0)],
+            dtype=jigsawpy.jigsaw_msh_t.EDGE2_t)
+
+        # flood fill to remove island, icecaps, etc.
+        searchedMask = np.zeros(sz)
+        floodMask = np.zeros(sz)
+        iStart = sz[0] // 2
+        jStart = sz[1] // 2
+        floodMask[iStart, jStart] = 1
+
+        neighbors = np.array([[1, 0], [-1, 0], [0, 1], [0, -1]])
+
+        lastSearchList = np.ravel_multi_index([[iStart], [jStart]],
+                                              sz, order='F')
+
+        # flood fill -------------------
+        cnt = 0
+        while len(lastSearchList) > 0:
+            # print(cnt)
+            cnt += 1
+            newSearchList = np.array([], dtype='i')
+
+            for iii in range(len(lastSearchList)):
+                [i, j] = np.unravel_index(lastSearchList[iii], sz, order='F')
+                # search neighbors
+                for n in neighbors:
+                    ii = i + n[0]
+                    jj = j + n[1]  # subscripts to neighbor
+                    # only consider unsearched neighbors
+                    if searchedMask[ii, jj] == 0:
+                        searchedMask[ii, jj] = 1  # mark as searched
+
+                        if thk[ii, jj] > 0.0:
+                            floodMask[ii, jj] = 1  # mark as ice
+                            # add to list of newly found  cells
+                            newSearchList = np.append(newSearchList,
+                                                      np.ravel_multi_index(
+                                                          [[ii], [jj]], sz,
+                                                          order='F')[0])
+            lastSearchList = newSearchList
+        # optional - plot flood mask
+        # plt.pcolor(floodMask)
+        # plt.show()
+
+        # apply flood fill
+        thk[floodMask == 0] = 0.0
+        vx[floodMask == 0] = 0.0
+        vy[floodMask == 0] = 0.0
+
+        # make masks -------------------
+        neighbors = np.array([[1, 0], [-1, 0], [0, 1], [0, -1],
+                              [1, 1], [-1, 1], [1, -1], [-1, -1]])
+
+        iceMask = thk > 0.0
+        # groundedMask = thk > (-1028.0 / 910.0 * topg)
+        # floatingMask = np.logical_and(thk < (-1028.0 /
+        #                               910.0 * topg), thk > 0.0)
+        marginMask = np.zeros(sz, dtype='i')
+        for n in neighbors:
+            marginMask = np.logical_or(marginMask,
+                                       np.logical_not(
+                                           np.roll(iceMask, n, axis=[0, 1])))
+        # where ice exists and neighbors non-ice locations
+        marginMask = np.logical_and(marginMask, iceMask)
+        # optional - plot mask
+        # plt.pcolor(marginMask); plt.show()
+
+        # calc dist to margin -------------------
+        [XPOS, YPOS] = np.meshgrid(x1, y1)
+        distToEdge = np.zeros(sz)
+
+        # -- KEY PARAMETER: how big of a search 'box' (one-directional) to use.
```

Comment:
Elaborate on this comment - what are we searching for?  (I realize you maybe didn't write this and maybe I did, but the details are foggy even to me...)

Reply:
That's a good question. I didn't write this, but it looks like this is finding the distance from every ice-filled cell to the nearest ice-sheet margin (distToEdge), which is then used as part of the mesh density function. I think the windowSize parameter is the length scale over which the distance from the ice-sheet margin affects the density function. But, as written this is pretty difficult to decipher. Not sure if we should just comment it much more thoroughly, or have a stab at rewriting the code to be more intuitive.

Reply:
Okay, I've improved this comment.

Reply:
This calculation of the distance to the ice edge could probably also be separated out into its own function for general use.